File: src/Moryx.Workplans.Web/src/app/components/sessions/editor/position.ts
/*
 * Copyright (c) 2026 Phoenix Contact GmbH & Co. KG
 * Licensed under the Apache License, Version 2.0
*/

export class Position {
    /**
     *
     */
    constructor(public left: number, public top: number) {        
    }

    move(x: number, y: number) {
        this.left += x;
        this.top += y;
    }
}

File: src/Moryx.Factory/Position.cs
// Copyright (c) 2026 Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using System.Runtime.Serialization;

namespace Moryx.Factory
{
    /// <summary>
    /// Position of a resource/machine
    /// </summary>
    [DataContract]
    public class Position
    {
        [DataMember]
        public double PositionX { get; set; }

        [DataMember]
        public double PositionY { get; set; }
    }
}